app/scripts/modules/kubernetes/securityGroup/configure/wizard/upsert.controller.js

```diff
@@ -181,6 +182,16 @@ module.exports = angular.module('spinnaker.securityGroup.kubernetes.create.contr
             cloudProvider: 'kubernetes',
             region: $scope.securityGroup.namespace,
           };
+
+          // Write TLS property so clouddriver understands it and our certs get configured
+          $scope.securityGroup.tls = [];
+          for (var idx in $scope.securityGroup.rules) {
+            let rule = $scope.securityGroup.rules[idx];
+            if (typeof rule.secret !== 'undefined' && rule.secret !== '') {
```

Comment:
Reading this makes me think that other JS false values might be acceptable here. If that's the case, I would make that explicit, e.g., 
```
if (rule.secret || rule.secret === null || rule.secret === 0 || rule.secret === NaN)  { ... }
```
otherwise, I would just stick with `if (rule.secret) { ... }`.